docs/site/migration/auth/example.md

```diff
@@ -0,0 +1,570 @@
+  make decision, only compares with the policies in the admin policy file.
+
+To make the tutorial concise, the code details is omitted here. You can find the
+ACLs and how the endpoints are decorated in file
+[src/controller/project.controller.ts](https://github.com/strongloop/loopback-next/blob/example/acl-migration/examples/access-control-migration/src/controllers/project.controller.ts)
+
+Then we write the authorizer that calls casbin enforcers to make decision
+
+(Since the PR is still in review, I will update the part when we agree on the
+details)
+
+- Create an authorizer that retrieves the authorization metadata from context,
+  then execute casbin enforcers to make decision
+- Create a voter for instance level endpoints to append the project id to the
+  resource name
+- Create casbin enforcers
+- Write casbin model and policies
+- Casbin persistency and synchronize(2nd Phase implementation, TBD)
+  - when create a new project
+    - create a set of p, project\${id}\_owner, action policies
+    - create a set of p, project\${id}\_team, action policies
+  - add a new user to a team
+    - find the projects owned by the team owner, then create role inherit rules
+      g, u${id}, project${id}\_team
+  - add a new endpoint(operation)
+    - for each of its allowed roles, add p, \${role}, action policy
+
+## Seed Data
+
+The application has pre-created data to try each role's permission. The original
+example seeds data in the boot script, now they are migrated to an observer file
+called `sample.observer.ts`.
+
+_Since the data are already generated in `db.json`, that observer file is
+skipped by default._
+
+## Try Out
+
+(a question for reviewers: do you prefer to see all the screenshots in this
+section? To be concise I tend to only describe by words, while if the flow is
+not clear enough, I can add pictures)
+
+Start the application:
+
+- Run `npm start` to start the application.
+- Open the explorer
+
+Try role 'admin':
+
+- Login as admin first (user Bob)
+  - Go to UserController, try endpoint `users/login` with {"email":
+    "bob@projects.com", "password": "opensesame"}
+  - Get the returned token
+  - Click authorize button and paste the token
+- Try the 5 endpoints, 'show-balance' and 'withdraw' will return 401, others
+  succeed
+
+Try role 'owner':
+
+- Login as owner (user John)
+  - Go to UserController, try endpoint `users/login` with {"email":
+    "John@projects.com", "password": "opensesame"}
+  - Get the returned token
+  - Click authorize button and paste the token
+- Try the 5 endpoints, 'view-all' will return 401, others succeed
+
+Try role 'team-member':
+
+- Login as team-member (user Jane)
+  - Go to UserController, try endpoint `users/login` with {"email":
+    "Jane@projects.com", "password": "opensesame"}
+  - Get the returned token
+  - Click authorize button and paste the token
+- Try the 5 endpoints, 'show-balance' and 'donate' will succeed, others returns
+  401
+
+## References
```

Comment:
I'm wondering whether it's necessary to show the CLI commands for each of them instead of just linking them to files?

Reply:
This is why I put the cli in reference :) if they just copy the file, they can ignore this part.